Core/Src/DynamicControls/lateral_control.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * lateral_control.c
+ *
+ *  Created on: 18 de ago de 2021
+ *      Author: Luiza
+ */
+
+#include "DynamicControls/lateral_control.h"
+#include "DynamicControls/initializer_controls.h"
+#include "DynamicControls/PID.h"
+#include "cmsis_os.h"
+#include "math.h"
+
+extern volatile float g_wheel_speed[4];
+extern volatile uint16_t steering_wheel;
+extern volatile uint8_t internal_wheel;
+extern volatile uint16_t gyro_yaw; // ainda nao existe
```

Comment:
acho melhor vc n buscar esses valores globalmente. Se n fosse RTOS, até q tudo bem, mas como a gnt tá fazendo em RTOS, esses valores podem mudar a qualquer momento, e se vc precisa deles de maneira “crítica”, vc n pode só pegar, vc vai passar a precisar sincronizar eles, com mutexes e semáforos. Eu to discutindo com o Telles uma maneira de contornar isso, mas por enquanto o q eu recomendo vc fazer é receber esses valores por parâmetro na sua função, a `lateral_control`, q aí vc passa os valores lá de onde vc vai chamar ela. Assim vc evita essa questão da sincronização.